src/backend/devices/kpro/kpro.py

```diff
@@ -12,7 +12,7 @@
 
 class Kpro:
```

Comment:
this deserves its own file with its own logic for the s300, it's scary to merge this without propper unit testing, may work for s300 but not sure if it will do for kpro anymore.

Reply:
Understood. I can work on breaking my stuff out into an S300 class if you want to work on something like a device handler? it would come in handy if we get our hands on a NepTune/Demmon board at some point since that is going to use an FTDI or direct serial connection.

Reply:
sure, that's it